docs note: `azurerm_cognitive_account`, not all `kind` support `storage` block (#24191)
hashicorp/terraform-provider-azurerm: website/docs/r/cognitive_account.html.markdown

* `identity_client_id` - (Optional) The client ID of the managed identity associated with the storage resource.

> ~> **NOTE:** Not all `kind` support a `storage` block. For example the `kind` `OpenAI` does not support it.

Reviewer: @zioproto is there some MSFT documentation that we can link here with a definitive list of what cognitive account kinds support storage? If not then I think we should rephrase this to be more specific
```suggestion
~> **NOTE:** Cognitive Accounts with `kind` `OpenAI` do not support a `storage` block.
```

Author: @felixwa Are you able to help me with this ask ? Is it just `OpenAI` that does not support this ? Thanks

Author: @stephybun it seems there is not a MSFT documentation page to link that publishes this definitive list. Should we just keep the generic sentence ?
```
Not all `kind` support a `storage` block.
```

I added the OpenAI example because it is very popular and a lot of customers ask about what the storage block is for.

Please advice what changes are necessary to get this merged. Thanks
